### PR TITLE
Use <relative-time> element to display time

### PR DESCRIPTION
**Pull request by theskumar**
This use https://github.com/github/relative-time-element to make the
display of the datetime accessible, localized, concise and standard
across the site.

This also make use of the custom elements, which is also used by
wagtail 4 at different places in the CMS

This PR also setup a way for us to use ESM modules.

## Screenshots (Before):

Screenshot 2023-05-16 at 08 29 59

Screenshot 2023-05-16 at 08 29 40

Screenshot 2023-05-16 at 08 30 23

## Screenshots (after):
Screenshot 2023-05-16 at 08 27 13

Screenshot 2023-05-16 at 08 28 02

Screenshot 2023-05-16 at 08 28 47

Screenshot 2023-05-16 at 08 29 07

Screenshot 2023-05-16 at 08 30 53

### CODE REVIEW

File: hypha/templates/base-apply.html [hunk 1]
```diff
@@ -47,6 +47,9 @@
         <script defer src="{% static 'js/apply/vendor/alpine-3.11.1.min.js' %}"></script>
         <style> [x-cloak] {display: none !important} </style>
         <!-- alpine.js end -->
+
+        <!-- modules -->
+        <script type="module" src="{% static 'js/esm/github-relative-time-element-4-3-0.js' %}"></script>
```
Comment (frjo) [inline hypha/templates/base-apply.html]: Can we remove the version number from the file name? A bit cumbersome on updates otherwise.

**Review by frjo — CHANGES_REQUESTED** [review]
Solid improvement, just a small issue regarding version number in file name.

File: hypha/templates/base-apply.html (hunk 1 above)
Comment (theskumar) [reply hypha/templates/base-apply.html]: Done.

File: package.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
     "license": "GPL-2.0",
     "babel": {
         "exclude": [
-            "**/*.min.js"
+            "**/*.min.js",
+            "**/esm/*.js"
         ],
         "presets": [
             [
```
Comment (frjo) [inline package.json:13]: Need to add `@github/relative-time-element` to `devDependencies`.

Comment (frjo) [reply package.json:13]: If we select to use npm install and copy-npm-scripts instead of download-esm.

**Review by frjo — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (theskumar) [reply package.json:13]: Take a look at this post for why we can't use npm for it (yet!), also why simply downloading packages from the cdn will also not work for esm modules. 

https://simonwillison.net/2023/May/2/download-esm/

Comment (theskumar) [reply package.json:13]: I can add more info if you like to the `download-esm-modules` target to explain why and when to use it. Let me know. 

It's currently supposed to be used manually and rename the files (to remove the version) or if we keep the version number in the html, then update the version number in the html.

Comment (frjo) [reply package.json:13]: Good arguments for `download-esm`. Can we have the make command rename the file? Printing out a messages about the need of renaming would work as well.

Renaming it manually is not a big thing, but may confuse a new developer, or us in a few month time :-).

Comment (theskumar) [reply package.json:13]: Added some notes in Makefile and base_apply.html on how to work this setup.

**Comment by theskumar** [comment]
This PR content is already merged to `main` so closing this now.

**Pull request closed without merging**